web/models.py

```
@@ -720,6 +723,16 @@ def youtube_id(self):
 
         return None
 
+    def save(self, *args, **kwargs):
+        if not self.video_id:
+            # Keep generating until a unique ID is found
+            while True:
+                new_id = uuid.uuid4().hex[:12]
+                if not EducationalVideo.objects.filter(video_id=new_id).exists():
+                    self.video_id = new_id
+                    break
+        super().save(*args, **kwargs)
```

Ruff (0.8.2)

Order of model's inner classes, methods, and fields does not follow the Django Style Guide: `save` method should come before custom method

(DJ012)